src/source-af-packet.c

```diff
@@ -1686,6 +1745,21 @@ static int AFPSetupRing(AFPThreadVars *ptv, char *devname)
     }
 #endif
 
+    /* Let's reserve head room so we can add the VLAN header in IPS
+     * or TAP mode before write the packet */
+    if (ptv->copy_mode != AFP_COPY_MODE_NONE) {
+        /* Only one vlan is extracted from AFP header so
+         * one VLAN header length is enough. */
+        int reserve = VLAN_HEADER_LEN;
+        if (setsockopt(ptv->socket, SOL_PACKET, PACKET_RESERVE, (void *) &reserve,
```

Comment:
Do you know when this feature was introduced? If this breaks existing AF_PACKET IPS setups we can't take it into 3.2.1. AFAICS this reserve is also done if no VLANS are on the network.